test/server.test.ts

```diff
@@ -1,36 +1,107 @@
 import { ApolloServer } from "@apollo/server";
 import fs from "fs";
-import { mockTransactions, resolvers } from "../src/resolvers";
+import { resolvers } from "../src/resolvers";
 import assert from "assert";
+import { MongoMemoryServer } from "mongodb-memory-server";
+import { MongoClient, Db, Collection, Document } from "mongodb";
+import AnalyticsDataSource from "../src/datasources/analytics";
+import { seedCollTxns } from "./seed/seed";
+import { ContextValue } from "../src/server";
+import { TransactionBatch } from "../src/types/Transaction";
 
 const typeDefs = fs.readFileSync("src/schema.graphql", "utf8");
+const KEY_HIDDEN_OBJECT_ID = "_id";
+
+let testDbServer: MongoMemoryServer;
+let testDbClient: MongoClient;
+let testDb: Db;
+let testCollTxns: Collection<Document>;
+let testServer: ApolloServer<ContextValue>;
+let testAnalyticsDS: AnalyticsDataSource;
 
 describe("server", () => {
-  describe("transactions", () => {
-    it("returns the mocked data of the correct format", async () => {
-      const server = new ApolloServer({
+  beforeAll(async () => {
+    try {
+      // setup test database
+      testDbServer = await MongoMemoryServer.create();
+      const client = await new MongoClient(testDbServer.getUri());
+      testDbClient = await client.connect();
+      testDb = testDbClient.db(AnalyticsDataSource.dbName);
+      testCollTxns = testDb.collection(AnalyticsDataSource.collTxns);
+      await testCollTxns.insertMany(seedCollTxns);
+
+      // setup test analytics datasource
+      testAnalyticsDS = await new AnalyticsDataSource(testDbServer.getUri());
+      await testAnalyticsDS.initialize();
+
+      // setup test server
+      testServer = new ApolloServer<ContextValue>({
```

Comment:
Setting up an instance of MongoDB, data source (wrapping MongoDB client), and Apollo server before any test case runs.